.github/workflows/benchmark.yml

```diff
@@ -52,60 +52,61 @@ jobs:
     timeout-minutes: 30
     outputs:
       image-tag: ${{ steps.image-tag.outputs.image-tag }}
+    permissions:
+      contents: 'read'
+      id-token: 'write'
     steps:
       - uses: actions/checkout@v3
         with:
           ref: ${{ inputs.ref }}
       - uses: google-github-actions/auth@v1
+        id: auth
         with:
-          credentials_json: '${{ secrets.GOOGLE_CREDENTIALS }}'
-      - uses: google-github-actions/setup-gcloud@v1
-        with:
-          project_id: zeebe-io
-          install_components: gke-gcloud-auth-plugin, kubectl
-      - run: gcloud auth configure-docker
-      - uses: ./.github/actions/setup-zeebe
-        with:
-          maven-cache: 'true'
-          secret_vault_secretId: ${{ secrets.VAULT_SECRET_ID }}
-          secret_vault_address: ${{ secrets.VAULT_ADDR }}
-          secret_vault_roleId: ${{ secrets.VAULT_ROLE_ID }}
+          token_format: 'access_token'
+          workload_identity_provider: 'projects/628707732411/locations/global/workloadIdentityPools/zeebe-gh-actions/providers/gha-provider'
+          service_account: 'zeebe-gh-actions@zeebe-io.iam.gserviceaccount.com'
```

Comment:
:question: Can you explain how this works without any secret/token/etc. form of authentication? Since this job will run on the GHA hosted runners, I'm a little confused as to how it authenticates itself. How is this restricted so not everyone can just run this?

Reply:
https://cloud.google.com/blog/products/identity-security/enabling-keyless-authentication-from-github-actions
looks like a good resource to dig deeper. 

My understanding is this: GitHub creates a token that includes the repository name & organization. I've setup the federation on Google's side so that it trusts tokens issues by GitHub for the zeebe-io and camunda repositories. 

This is not the first time we used federation btw, see https://github.com/camunda/zeebe-e2e-test/blob/main/.github/workflows/publish.yml#L23-L29